core/src/main/java/hudson/model/UserProperty.java

```diff
@@ -75,6 +77,10 @@ public UserPropertyDescriptor getDescriptor() {
     }
 
     public UserProperty reconfigure(StaplerRequest req, JSONObject form) throws FormException {
-        return form==null ? null : getDescriptor().newInstance(req, form);
+        if (form==null) {
+            return null;
+        }
+        UserPropertyDescriptor descriptor = getDescriptor();
+        return descriptor==null ? null : descriptor.newInstance(req, form);
```

Comment:
if the descriptor is null, I think you should just return this (ie keep the property so that it is not lost)

Reply:
Thanks!